aea/connections/stub/connection.py

```diff
@@ -57,20 +57,25 @@ def _encode(e: Envelope, separator: bytes = SEPARATOR):
     result += str(e.protocol_id).encode("utf-8")
     result += separator
     result += e.message
+    result += separator
 
     return result
 
 
 def _decode(e: bytes, separator: bytes = SEPARATOR):
-    split = e.split(separator, maxsplit=3)
+    split = e.split(separator, maxsplit=4)
```

Comment:
@marcofavorito I had to change this as the bytes encoded strings could contain new line characters rendering the old solution broken.